assets/js/hooks/usePrevious.tsx

```diff
@@ -0,0 +1,9 @@
+import { useEffect, useRef } from "react";
+
+export function usePrevious<T>(value: T) {
```

Comment:
If you want to do the usual arrow function style, this can be written like:
```ts
const usePrevious = <T,>(value: T) => {
  ...
};
```

(Note the comma, which lets the parser rule out the `<T>` as a JSX expression)